frontend/src/components/User/UserProfile.tsx

```diff
@@ -31,15 +32,26 @@ const UserProfile = ({ id }: UserProfileProps) => {
   const [currentTab, setCurrentTab] = useState('Home');
 
   const determineTabContent = () => {
-    switch (currentTab) {
-      case 'Home':
+    if (currentTab === 'Home') {
+      if (loadingUserBlogs) {
+        return <Spinner />;
+      } else if (!blogs || blogs.length === 0) {
+        return (
+          <div className="empty-state-container">
+            <div className="empty-state-message">You have not written any articles yet.</div>
+            <a href="/publish" className="write-blog-button">Write Blog</a>
+          </div>
```

Comment:
Create a new stateless component for the jsx (you can keep the component within this file) and call the component here instead of mixing Jsx with javascript.

Also, there is no need to change switch to if-else statements. Since it's possible to achieve the results using switch as well. Is there any specific reason why it was changed ?